core/mail_api.php

```diff
@@ -1316,9 +1316,18 @@ private function mail_is_a_bugnote( $p_mail_subject, $p_references )
 		//Get the ids from Mail References(header)
 		$t_bug_id = $this->get_bug_id_from_references( $p_references );
 
-		if ( $t_bug_id !== FALSE && bug_exists( $t_bug_id ) )
+		if ( $t_bug_id !== FALSE )
 		{
-			return( $t_bug_id );
+			if( bug_exists( $t_bug_id ) )
+			{
+				return( $t_bug_id );
+			}
+			else
+			{
+				// We found a referenced bug_id that does not exists.
+				// Do a clean up of the table to avoid inconsistencies.
+				self::clean_references_for_deleted_issues();
```

Comment:
related to this: https://github.com/mantisbt/mantisbt/pull/974
the delete event is not triggered consistently (in that case, deleting a project with all its issues)
also, soap api delete does not call the event
i was also thinking of these scenarios:
external code that calls bug_delete(), but not calls the event.
manual deleting from database

Even it said PR gets it fixed, it will be consistent only from version>1.3.5/6